game-core/src/main/java/games/strategy/net/PlayerNameAssigner.java

```diff
@@ -0,0 +1,83 @@
+package games.strategy.net;
+
+import java.net.InetAddress;
+import java.util.Collection;
+import java.util.Optional;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Utility class that will assign a name to a newly logging in player.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class PlayerNameAssigner {
+
+  /**
+   * Name string returned in the case of a hacked client sending us an invalid/unexpected name.
+   */
+  @VisibleForTesting
+  static final String DUMMY_NAME = "aa";
+
+  /**
+   * Returns a node name, based on the specified node name, that is unique across all nodes. The node name is made
+   * unique by adding a numbered suffix to the existing node name. For example, for the second node with the name "foo",
+   * this method will return "foo (1)".
+   *
+   * @param desiredName The name being requested by a new user joining.
+   * @param ipAddress The IP address of the user that is joining. Used to determine if the user
+   *        is already connected under a different name.
+   * @param loggedInNodes Collection of nodes that have already joined.
+   */
+  public static String assignName(
+      final String desiredName,
+      final InetAddress ipAddress,
+      final Collection<INode> loggedInNodes) {
+    Preconditions.checkNotNull(ipAddress);
+    Preconditions.checkNotNull(loggedInNodes);
+
+    String currentName =
+        (desiredName == null || desiredName.length() < 3)
+            ? DUMMY_NAME
+            : findLoggedInName(ipAddress, loggedInNodes).orElse(desiredName);
+    if (currentName.length() > 50) {
+      currentName = currentName.substring(0, 50);
+    }
+    if (currentName.length() < 2) {
+      currentName = "aa" + currentName;
```

Comment:
Does this prefix have something to do with `DUMMY_NAME`?
If not, could you document why?

Reply:
This if statement should have been just removed: b9513050d90623449b309747dd560c19f65ae93f

We are defending only against custom clients here. I'm not sure offhand what the best way to handle this is, maybe even generating a random name or probably more properly to disallow the name.

I went ahead and made that update here, to reject names that are too short and that removes the need for any 'dummy' name code: 765cf0d6a

Small nuance is that typically the too short name case should not happen, but we can have a custom client that forces the scenario, so server should deal with it.